### PR TITLE
Qemu/armv7: use Arm-TF as secure bootloader

### PR DESCRIPTION
**Pull request by etienne-lms**
These changes allows to build arm-tf for qemu_virt (cortex-a15), build bios_qemu_tz_arm to act as a non secure bootloader and boot the qemu from arm-tf BL1/BL2.

Main makefile currently rely on directive `USE_ATF=y/n` to enable boot from arm-tf.
When booting qemu from arm-tf, one should set `CFG_QEMU_ATF_MAILBOX=y` for Qemu to wakeup secondary cores (case `QEMU_SMP>1` or `CFG_TEE_CORE_NB_CORE>1`).

If the arm-tf boot becomes to standard qemu boot scheme, `USE_ATF=y` will be implicit, and `CFG_QEMU_ATF_MAILBOX=y` will be default enable from the op-tee platform. 

Requires:
  https://github.com/linaro-swg/bios_qemu_tz_arm/pull/15
  https://github.com/OP-TEE/optee_os/pull/2138

### CODE REVIEW

File: qemu.mk [hunk 1]
```diff
@@ -157,12 +217,38 @@ update_rootfs: update_rootfs-common
...
 .PHONY: run
 # This target enforces updating root fs etc
 run: all
 	$(MAKE) run-only
 
 .PHONY: run-only
+
+ifeq ($(USE_ATF),y)
+run-only: get-binaries
+get-binaries:
```
Comment (jenswi-linaro) [inline qemu.mk]: Should have `.PHONY` attribute for `get-binaries`

File: qemu.mk [hunk 2]
```diff
@@ -157,12 +217,38 @@ update_rootfs: update_rootfs-common
...
+	ln -sf $(ARM_TF_OUT)/bl1.bin $(BINARIES_PATH)/bl1.bin
+	ln -sf $(ARM_TF_OUT)/bl2.bin $(BINARIES_PATH)/bl2.bin
+	ln -sf $(ROOT)/out/bios-qemu/bios.bin $(BINARIES_PATH)/bl33.bin
+	ln -sf $(OPTEE_OS_HEADER_V2_BIN) $(BINARIES_PATH)/bl32.bin
+	ln -sf $(OPTEE_OS_PAGER_V2_BIN) $(BINARIES_PATH)/bl32_extra1.bin
+	ln -sf $(OPTEE_OS_PAGEABLE_V2_BIN) $(BINARIES_PATH)/bl32_extra2.bin
+
+BIOS_PATH := $(BINARIES_PATH)/bl1.bin
+else
+BIOS_PATH := $(ROOT)/out/bios-qemu/bios.bin
```
Comment (jenswi-linaro) [inline qemu.mk]: Do we need to keep the old bios hack?

Comment (etienne-lms) [reply qemu.mk]: Hmm, no. Actually, this p-r is almost a rfc. This is why I kept both. But in the end, we should use only 1 scheme. I'll update the p-r, discarding the old boot scheme support.

**Comment by etienne-lms** [comment]
This needs an update.... on going.

**Comment by etienne-lms** [comment]
Dependencies updated:
  https://github.com/linaro-swg/bios_qemu_tz_arm/pull/15
  https://github.com/OP-TEE/optee_os/pull/2138

Well... there is still an added dependency: one shall clone the arm-trusted-firmware to get it built!

**Pull request closed without merging**